src/app/pages/Swap/form/EvmSwapForm/index.tsx

```diff
@@ -0,0 +1,503 @@
+import { t } from 'lib/i18n';
+import { getAssetSymbol, useGetEvmChainAssetMetadata } from 'lib/metadata';
+import { atomsToTokens, tokensToAtoms } from 'lib/temple/helpers';
+import { isEvmNativeTokenSlug } from 'lib/utils/evm.utils';
+import { toBigInt, ZERO } from 'lib/utils/numbers';
+import { getViemPublicClient } from 'temple/evm';
+import { useAccountForEvm } from 'temple/front';
+import { useEvmChainByChainId } from 'temple/front/chains';
+
+import { EvmReviewData } from '../interfaces';
+
+interface ChainAssetInfo {
+  networkName: string;
+  chainId: number;
+  assetSlug: string;
+}
+
+interface EvmSwapFormProps {
+  chainId: number;
+  slippageTolerance: number;
+  onReview: SyncFn<EvmReviewData>;
+  onSelectAssetClick: (field: 'from' | 'to') => void;
+  selectedChainAssets: { from: string | null; to: string | null };
+  activeField: 'from' | 'to';
+  handleToggleIconClick: EmptyFn;
+}
+
+const DEBOUNCE_RATE_MS = 200;
+const AUTO_REFRESH_INTERVAL_MS = 30000; // 30 seconds
+
+export const EvmSwapForm: FC<EvmSwapFormProps> = ({
+  chainId,
+  slippageTolerance,
+  onReview,
+  onSelectAssetClick,
+  selectedChainAssets,
+  activeField,
+  handleToggleIconClick
+}) => {
+  const account = useAccountForEvm();
+  const network = useEvmChainByChainId(chainId);
+  if (!account || !network) throw new DeadEndBoundaryError();
+
+  const publicKeyHash = account.address as HexString;
+
+  const [swapRoute, setSwapRoute] = useState<Route | null>(null);
+  const [isRouteLoading, setIsRouteLoading] = useState(false);
+  const lifiStep = useMemo(
+    () => (swapRoute?.steps?.[0]?.type === 'lifi' ? swapRoute?.steps[0] : undefined),
+    [swapRoute]
+  );
+
+  const [isAlertVisible, setIsAlertVisible] = useState(false);
+
+  const getTokenMetadata = useGetEvmChainAssetMetadata(chainId);
+  const formAnalytics = useFormAnalytics('SwapForm');
+
+  const sourceAssetInfo = useMemo<ChainAssetInfo | null>(() => {
+    if (!selectedChainAssets.from) return null;
+
+    const [networkName, chainId, assetSlug] = parseChainAssetSlug(selectedChainAssets.from);
+    return {
+      networkName,
+      chainId: Number(chainId),
+      assetSlug
+    };
+  }, [selectedChainAssets.from]);
+
+  const targetAssetInfo = useMemo<ChainAssetInfo | null>(() => {
+    if (!selectedChainAssets.to) return null;
+
+    const [networkName, chainId, assetSlug] = parseChainAssetSlug(selectedChainAssets.to);
+    return {
+      networkName,
+      chainId: Number(chainId),
+      assetSlug
+    };
+  }, [selectedChainAssets.to]);
+
+  const defaultValues = useMemo(() => {
+    return {
+      input: {
+        assetSlug: sourceAssetInfo?.assetSlug
+      },
+      output: {
+        assetSlug: targetAssetInfo?.assetSlug
+      },
+      isFiatMode: false
+    };
+  }, [sourceAssetInfo, targetAssetInfo]);
+
+  const form = useForm<SwapFormValue>({
+    defaultValues,
+    mode: 'onSubmit',
+    reValidateMode: 'onChange'
+  });
```

Comment:
declare getDefaultSwapFormValues fn in Swap/form/utils.ts

```suggestion
  const form = useForm<SwapFormValue>({
    defaultValues: getDefaultSwapFormValues(sourceAssetInfo?.assetSlug, targetAssetInfo?.assetSlug),
    mode: 'onSubmit',
    reValidateMode: 'onChange'
  });
```